### ISSUE TITLE
[1.5.2, static libs] Pod install triggers recompile of all pods

### ISSUE THREAD

**Issue by MaximeLM** [CocoaPods/CocoaPods#7787]
* [X] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report 🌈

## What did you do?

* Run `pod install`, open project, build
* Close project, run `pod install`, open project
* Build project

## What did you expect to happen?

Xcode should not recompile pods because no file has changed.

## What happened instead?

* When using `use_frameworks!`, pods are not recompiled
* When removing `use_frameworks!`, pods are recompiled

### Stack

```
   CocoaPods : 1.5.2
        Ruby : ruby 2.3.3p222 (2016-11-21 revision 56859) [universal.x86_64-darwin17]
    RubyGems : 2.5.2
        Host : Mac OS X 10.13.4 (17E202)
       Xcode : 9.3.1 (9E501)
         Git : git version 2.15.1 (Apple Git-101)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ ff10e1c3fc6c8d8b784073a440c51220edc308c2
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
platform :osx, '10.10'

target 'TestPodInstall' do
  #use_frameworks!

  pod 'RxSwift'

end
```

## Project that demonstrates the issue

https://github.com/MaximeLM/TestPodInstall

**Comment by dnkoutso** [comment]
Can you expand a compilation command in the output log, I believe Xcode has a parenthesis saying why a re-build was triggered for a file.

**Comment by MaximeLM** [comment]
I don't know what to look for, here is the output log for a single file:
```
CompileSwift normal x86_64 /Users/maximelemoine/Downloads/TestPodInstall-master/Pods/RxSwift/RxSwift/Observables/AddRef.swift
    cd /Users/maximelemoine/Downloads/TestPodInstall-master/Pods
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift -frontend -c -filelist /var/folders/v3/4r8_k73d42xcf35yvh8m7wsw0000gn/T/sources-fa013b -primary-file /Users/maximelemoine/Downloads/TestPodInstall-master/Pods/RxSwift/RxSwift/Observables/AddRef.swift -target x86_64-apple-macosx10.10 -enable-objc-interop -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -I /Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Products/Debug/RxSwift -F /Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Products/Debug/RxSwift -enable-testing -g -import-underlying-module -module-cache-path /Users/maximelemoine/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -swift-version 4 -enforce-exclusivity=checked -Onone -D DEBUG -D COCOAPODS -serialize-debugging-options -Xcc -fmodule-map-file=/Users/maximelemoine/Downloads/TestPodInstall-master/Pods/Headers/Public/RxSwift/RxSwift.modulemap -Xcc -I/Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Intermediates.noindex/Pods.build/Debug/RxSwift.build/swift-overrides.hmap -Xcc -iquote -Xcc /Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Intermediates.noindex/Pods.build/Debug/RxSwift.build/RxSwift-generated-files.hmap -Xcc -I/Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Intermediates.noindex/Pods.build/Debug/RxSwift.build/RxSwift-own-target-headers.hmap -Xcc -I/Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Intermediates.noindex/Pods.build/Debug/RxSwift.build/RxSwift-all-target-headers.hmap -Xcc -iquote -Xcc /Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Intermediates.noindex/Pods.build/Debug/RxSwift.build/RxSwift-project-headers.hmap -Xcc -I/Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Products/Debug/RxSwift/include -Xcc -I/Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Intermediates.noindex/Pods.build/Debug/RxSwift.build/DerivedSources/x86_64 -Xcc -I/Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Intermediates.noindex/Pods.build/Debug/RxSwift.build/DerivedSources -Xcc -DPOD_CONFIGURATION_DEBUG=1 -Xcc -DDEBUG=1 -Xcc -DCOCOAPODS=1 -Xcc -working-directory/Users/maximelemoine/Downloads/TestPodInstall-master/Pods -emit-module-doc-path /Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Intermediates.noindex/Pods.build/Debug/RxSwift.build/Objects-normal/x86_64/AddRef~partial.swiftdoc -serialize-diagnostics-path /Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Intermediates.noindex/Pods.build/Debug/RxSwift.build/Objects-normal/x86_64/AddRef.dia -module-name RxSwift -emit-module-path /Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Intermediates.noindex/Pods.build/Debug/RxSwift.build/Objects-normal/x86_64/AddRef~partial.swiftmodule -emit-dependencies-path /Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Intermediates.noindex/Pods.build/Debug/RxSwift.build/Objects-normal/x86_64/AddRef.d -emit-reference-dependencies-path /Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Intermediates.noindex/Pods.build/Debug/RxSwift.build/Objects-normal/x86_64/AddRef.swiftdeps -o /Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Build/Intermediates.noindex/Pods.build/Debug/RxSwift.build/Objects-normal/x86_64/AddRef.o -index-store-path /Users/maximelemoine/Library/Developer/Xcode/DerivedData/TestPodInstall-dobqqrdaeijciifhubxrkvbjaycy/Index/DataStore -index-system-modules
```

And the entire log (RTF): [Build TestPodInstall_2018-05-28T17-28-45.txt](https://github.com/CocoaPods/CocoaPods/files/2047631/Build.TestPodInstall_2018-05-28T17-28-45.txt)

**Comment by dnkoutso** [comment]
@MaximeLM this might be an Xcode 9.x issue. I tried with 10 beta and a second build was instant.

<img width="1512" alt="screen_shot_2018-06-15_at_4_17_52_am" src="https://user-images.githubusercontent.com/310370/41465487-2c5ab6f6-7053-11e8-9aec-3482f9ef3040.png">

**Comment by segiddins** [comment]
Can you try #7842?

**Comment by MaximeLM** [comment]
@segiddins I just tried the pull request, the second was instant 👍

**Comment by segiddins** [comment]
Yay!

**Comment by dnkoutso** [comment]
Fix was merged for 1.6.0 with https://github.com/CocoaPods/CocoaPods/pull/7842. Going to close this.
